app/models/provider_user_notification.rb

```diff
@@ -0,0 +1,3 @@
+class ProviderUserNotification < ActiveRecord::Base
```

Comment:
Would be good to have `ProviderUserNotificationPreferences`! It reads a bit like it's a notification itself. I think you can do something like `self.table_name = :provider_user_notifications`. Or alternatively manually create the index in the migration and give it a shorter name there.